### PR TITLE
[IMP] add field channel_type on sale.channel

### PR DESCRIPTION
**Pull request by clementmbr**
Useful fields for the Amazon connector!

### CODE REVIEW

File: sale_channel/models/sale_channel.py [hunk 1]
```diff
@@ -10,3 +10,8 @@ class SaleChannel(models.Model):
 
     name = fields.Char(required=True)
     active = fields.Boolean(default=True)
+    type_channel = fields.Selection(
+        [], help="Allows to use specific fields and actions for specific channel's type"
+    )
```
Comment (florian-dacosta) [inline sale_channel/models/sale_channel.py:17]: I prefer 'channel_type' on my side, but I guess it is just a matter of taste :)

**Review by florian-dacosta — COMMENTED** [review]
LGTM but you need to add these in the view, no ?

File: sale_channel/models/sale_channel.py (hunk 1 above)
Comment (clementmbr) [reply sale_channel/models/sale_channel.py:17]: it can be channel_type for sure! :)

**Comment by clementmbr** [comment]
@florian-dacosta I'm thinking now that it may be better to add these fields in the sale_import_base module... What do you prefer?

**Comment by clementmbr** [comment]
@florian-dacosta I finally chose to put only `channel_type` in the base module sale_channel as I thought `company_id` or `crm_team_id` are more Import Parameters to be transferred to the new sale.order

**Review by florian-dacosta — APPROVED** [review]

**Review by bealdav — APPROVED** [review]
LGTM

**Review by adrienpeiffer — APPROVED** [review]
LGTM

**Comment by rousseldenis** [comment]
@sebastienbeau @clementmbr Could you check why connector_search_engine make tests red ?

**Review by timloic — APPROVED** [review]
That make sense

**Comment by clementmbr** [comment]
> @sebastienbeau @clementmbr Could you check why connector_search_engine make tests red ?

Hi @rousseldenis, [as you can see](https://github.com/OCA/sale-channel/actions/runs/8332403108/job/22801435634?pr=17#step:7:467) it's red because `connector_search_engine` is loading the external dependency [`typing_extensions`](https://pypi.org/project/typing-extensions/) which is [not in the module's manifest](https://github.com/OCA/search-engine/blob/16.0/connector_search_engine/__manifest__.py#L16).

I can open a PR on the search-engine repo to add the external dependency, but I wonder why, if this is the problem, there is no other broken tests on the repo.

Is it possible to re-run the tests first to see it can pass?
Thanks!

**Comment by clementmbr** [comment]
@rousseldenis @florian-dacosta @bealdav with https://github.com/OCA/search-engine/pull/194 merged, this PR can be merged!

Thanks!

**Review by rvalyi — APPROVED** [review]

**Pull request merged**